process/smartContract/hooks/blockChainHook.go

```diff
@@ -585,8 +585,16 @@ func (bh *BlockChainHookImpl) getUserAccounts(
 	input *vmcommon.ContractCallInput,
 ) (vmcommon.UserAccountHandler, vmcommon.UserAccountHandler, error) {
 	var sndAccount vmcommon.UserAccountHandler
+
 	sndShardId := bh.shardCoordinator.ComputeId(input.CallerAddr)
-	if sndShardId == bh.shardCoordinator.SelfId() {
+	selfShard := bh.shardCoordinator.SelfId()
+
+	isSelfShardSovereign := selfShard == core.SovereignChainShardId
```

Comment:
this will never be true. You have to set in the constructor of the blockchainHookImpl, whether this is a mainchain blockchainHook, or SovereignBlockChainHook.

Reply:
Created a sovereign blockchain hook as suggested